lightkurve/tests/test_lightcurve.py

```diff
@@ -173,6 +173,69 @@ def test_lightcurve_append_multiple():
     assert_array_equal(lc.time, 4*[1, 2, 3])
 
 
+def test_lightcurve_copy():
+    """Test ``LightCurve.copy()``."""
+    time = np.array([1, 2, 3, 4])
+    flux = np.array([1, 2, 3, 4])
+    error = np.array([0.1, 0.2, 0.3, 0.4])
+    lc = LightCurve(time=time, flux=flux, flux_err=error)
+
+    nlc = lc.copy()
+    assert_array_equal(lc.time, nlc.time)
+    assert_array_equal(lc.flux, nlc.flux)
+    assert_array_equal(lc.flux_err, nlc.flux_err)
+
+    np.put(nlc.time, 1, 5)
+    np.put(nlc.flux, 1, 6)
+    np.put(nlc.flux_err, 1, 7)
+
+    with pytest.raises(AssertionError, match='(mismatch 25.0%)'):
+        assert_array_equal(lc.time, nlc.time)
+
+    with pytest.raises(AssertionError, match='(mismatch 25.0%)'):
+        assert_array_equal(lc.flux, nlc.flux)
+
+    with pytest.raises(AssertionError, match='(mismatch 25.0%)'):
+        assert_array_equal(lc.flux_err, nlc.flux_err)
+
+    # KeplerLightCurve has extra data
+    lc = KeplerLightCurve(time=[1, 2, 3], flux=[1, .5, 1],
+                          centroid_col=[4, 5, 6], centroid_row=[7, 8, 9],
+                          cadenceno=[10, 11, 12], quality=[10, 20, 30])
+    nlc = lc.copy()
+    assert_array_equal(lc.time, nlc.time)
+    assert_array_equal(lc.flux, nlc.flux)
+    assert_array_equal(lc.centroid_col, nlc.centroid_col)
+    assert_array_equal(lc.centroid_row, nlc.centroid_row)
+    assert_array_equal(lc.cadenceno, nlc.cadenceno)
+    assert_array_equal(lc.quality, nlc.quality)
+
+    np.put(nlc.time, 1, 6)
+    np.put(nlc.flux, 1, 7)
+    np.put(nlc.centroid_col, 1, 8)
+    np.put(nlc.centroid_row, 1, 9)
+    np.put(nlc.cadenceno, 1, 10)
+    np.put(nlc.quality, 1, 11)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.time, nlc.time)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.flux, nlc.flux)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.centroid_col, nlc.centroid_col)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.centroid_row, nlc.centroid_row)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.cadenceno, nlc.cadenceno)
+
+    with pytest.raises(AssertionError, match='mismatch 33.3333333333'):
+        assert_array_equal(lc.quality, nlc.quality)
+
```

Comment:
How sensitive is the match to the number of decimal places?  If the Assertion error moves to nine decimal places instead of ten, will the tests fail?

Reply:
Thanks for the input! I did that at first, but it seemed like a huge run-on sentence, so I spaced it

Reply:
> How sensitive is the match to the number of decimal places? If the Assertion error moves to nine decimal places instead of ten, will the tests fail?

Not at all - so long as it is a partial match, it will work. As a result, the only issue would be matching too much, like `33.3333333337%` along with `33.3333333333%`

Reply:
> Unless you mean if the `AssertionError` was originally outputting 10 decimal places, changed to 9 decimal places, but we're still checking for 10 decimal places. In that case, yes it would fail because it did not match a number to 10 decimal places.

Resolved using regular expressions to detect any number of decimal places provided they are 3's.